kotlin-ktor/src/test/java/com/baeldung/graphql/client/GraphQLClientTest.kt

```diff
@@ -0,0 +1,142 @@
+        assertNull(fetchedConference)
+    }
+
+    @Test
+    fun `given create or save conference when saving not existing conference then should create it`(): Unit = runBlocking {
+        graphQLService.createOrSaveConference(null, "Some Conference")
+        val createdConference = ConferenceRepository.findById(0)
+        assertEquals("Some Conference", createdConference?.name)
+    }
+
+    @Test
+    fun `given create or save conference when saving existing conference then should update it`(): Unit = runBlocking {
+        createDummyConference()
+        graphQLService.createOrSaveConference(0, "Same Conference")
+        val updatedConference = ConferenceRepository.findById(0)
+        assertEquals("Same Conference", updatedConference?.name)
+    }
+
+    @Test
+    fun `given create or save attendee when saving not existing attendee then should create it`(): Unit = runBlocking {
+        graphQLService.createOrSaveAttendee(null, "Some Attendee")
+        val createdAttendee = AttendeeRepository.findById(0)
+        assertEquals("Some Attendee", createdAttendee?.name)
+    }
+
+    @Test
+    fun `given create or save attendee when saving existing attendee then should update it`(): Unit = runBlocking {
+        createDummyAttendee()
+        graphQLService.createOrSaveAttendee(0, "Same Attendee")
+        val updatedAttendee = AttendeeRepository.findById(0)
+        assertEquals("Same Attendee", updatedAttendee?.name)
+    }
+
+    @Test
+    fun `given get object by id when fetching conference then should return it`(): Unit = runBlocking {
+        createDummyConference()
+        val objectById = graphQLService.getObjectById(0)
+        assertIs<com.baeldung.graphql.client.generated.objectbyidquery.Conference>(objectById)
+    }
+
+    @Test
+    fun `given get object by id when fetching attendee then should return it`(): Unit = runBlocking {
+        createDummyAttendee()
+        val objectById = graphQLService.getObjectById(0)
+        assertIs<com.baeldung.graphql.client.generated.objectbyidquery.Attendee2>(objectById)
+    }
+
+    @Test
+    fun `given get object by id when fetching not existing object then should return null`(): Unit = runBlocking {
+        assertNull(graphQLService.getObjectById(0))
+    }
+
+    @Test
+    fun `given conference with attendees when fetching conference with limit one then should return conference with one attendee object`(): Unit = runBlocking {
+        createDummyAttendee()
+        createDummyAttendee()
+        createDummyConference(listOf(0, 1))
+        val fetchedConference = graphQLService.findConferenceById(0, 1)
+        assertEquals(1, fetchedConference?.attendeeObjects?.size)
+    }
+
+    @Test
+    fun `given conference batch when fetching two conferences then should return both`() = runBlocking {
+        createDummyConference()
+        createDummyConference()
+        val conferences = graphQLService.getConferenceBatch(0, 1)
```

Comment:
For clarity, can we use named parameters:
```suggestion
        val conferences = graphQLService.getConferenceBatch(firstId = 0, secondId = 1)
```

Reply:
✅